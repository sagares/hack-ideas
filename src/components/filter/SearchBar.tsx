import React from "react";

type SearchBarProps = {
  searchText: string;
  onFilter: (searchText: string) => void;
}

export const SearchBar = ({ searchText, onFilter }: SearchBarProps) => {
  const handleSearch = (e: React.FormEvent<HTMLInputElement>) => {
    onFilter(e.currentTarget.value);
  };

  const handleClear = () => {
    onFilter("");
  };

  return (
    <nav>
      <div className="nav-wrapper">
        <form>
          <div className="input-field grey lighten-4">
            <input
              id="search"
              type="search"
              value={searchText}
              onChange={handleSearch}
            />
            <label className="label-icon" htmlFor="search">
              <i className="material-icons">search</i>
            </label>
            <i className="clear-icon material-icons" onClick={handleClear}>
              close
            </i>
          </div>
        </form>
      </div>
    </nav>
  );
};
