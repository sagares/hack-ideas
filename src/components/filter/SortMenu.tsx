import React, { useEffect } from "react";
import M from "materialize-css";

type SortMenuProps = {
  sortBy: string;
  onSort: (sortBy: string) => void;
};

export const SortMenu = ({ sortBy, onSort }: SortMenuProps) => {
  useEffect(() => {
    var dropdowns = document.querySelectorAll(".dropdown-trigger");
    M.Dropdown.init(dropdowns, {});
  }, []);

  return (
    <React.Fragment>
      <a
        className="dropdown-trigger btn-floating btn-large"
        data-target="dropdown1"
      >
        <i className="large material-icons">sort</i>
      </a>

      <ul id="dropdown1" className="dropdown-content">
        <li className={`${sortBy === "submittedOn" ? "active" : ""}`}>
          <a onClick={() => onSort("submittedOn")}>
            <i className="material-icons">access_time</i>By Created On
          </a>
        </li>
        <li className={`${sortBy === "upvotes" ? "active" : ""}`}>
          <a onClick={() => onSort("upvotes")}>
            <i className="material-icons">thumb_up</i>By Upvotes
          </a>
        </li>
        <li className="divider" tabIndex={-1}></li>
        <li>
          <a onClick={() => onSort("")}>
            <i className="material-icons">cached</i>Reset
          </a>
        </li>
      </ul>
    </React.Fragment>
  );
};
