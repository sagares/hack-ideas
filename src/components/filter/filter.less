@import "../../colors.less";

.hack-ideas--filter {
    align-items: center;
    display: flex;
    justify-content: space-between;
    padding: 0.7em;
    margin-bottom: 0;

    .search-bar {
      .nav-wrapper {
        .input-field {
          &.grey {
            label {
              i {
                color: @body-text;
              }
            }
          }
        }
      }
      @media only screen and (min-width: 601px) {
        nav {
          .height50();
          .nav-wrapper {
            i {
              .height50();
            }
          }
        }
      }
    }
    .filter-dropdown {
      display: flex;
      margin: 0 !important;
      position: relative;

      .btn-large {
        background-color: @emphasis-clr;
        i {
          color: @dark-clr;
        }
      }
      .dropdown-content {
        height: auto !important;
        width: auto !important;
        li {
          > a {
            color: @emphasis-clr;
            display: flex;
          }
        }
      }
    }
  }

  .height50() {
    height: 50px;
    line-height: 50px;
  }