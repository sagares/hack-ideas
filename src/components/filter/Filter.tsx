import React from "react";
import { FilterProps } from "../../HackIdea.types";
import { SearchBar } from ".";
import { SortMenu } from "./SortMenu";

import "./filter.less";

const Filter = ({ searchText, onFilter, sortBy, onSort }: FilterProps) => {
  return (
    <div className="hack-ideas--filter row">
      <div className="search-bar col s12 m12 l4">
        <SearchBar searchText={searchText} onFilter={onFilter} />
      </div>
      <div className="filter-dropdown col s4 m2 l2">
        <SortMenu sortBy={sortBy} onSort={onSort} />
      </div>
    </div>
  );
};

export default Filter;
