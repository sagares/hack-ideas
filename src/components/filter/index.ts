import { SearchBar } from "./SearchBar";
import { SortMenu } from "./SortMenu";
import Filter from "./Filter";

export default Filter;
export { SearchBar, SortMenu }