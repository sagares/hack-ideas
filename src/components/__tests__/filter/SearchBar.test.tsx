/**
 * @jest-environment jsdom
 */
import React from "react";
import { fireEvent, render } from "@testing-library/react";
import { SearchBar } from "../../filter";

describe("Search Bar Component", () => {
  let element, component, onFilterSpy;

  beforeEach(() => {
    onFilterSpy = jest.fn();

    element = document.createElement("div");

    component = render(
      <SearchBar searchText="mock-search" onFilter={onFilterSpy} />,
      {
        container: document.body.appendChild(element),
      }
    );
  });

  afterEach(() => {
    document.body.removeChild(element);
  });

  test("renders", () => {
    expect(component.container).toMatchSnapshot();
  });

  test("filters when search input is changed", () => {
    const searchInput = element.querySelector("#search");
    fireEvent.change(searchInput, { target: { value: "foobar" } });
    expect(onFilterSpy).toHaveBeenCalledWith("foobar");
  });

  test("clears search when clear icon is clicked", () => {
    const clearIcon = element.querySelector(".clear-icon");
    fireEvent.click(clearIcon);
    expect(onFilterSpy).toHaveBeenCalledWith("");
  });
});
