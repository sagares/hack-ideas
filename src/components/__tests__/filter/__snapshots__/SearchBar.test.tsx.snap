// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar Component renders 1`] = `
<div>
  <nav>
    <div
      class="nav-wrapper"
    >
      <form>
        <div
          class="input-field grey lighten-4"
        >
          <input
            id="search"
            type="search"
            value="mock-search"
          />
          <label
            class="label-icon"
            for="search"
          >
            <i
              class="material-icons"
            >
              search
            </i>
          </label>
          <i
            class="clear-icon material-icons"
          >
            close
          </i>
        </div>
      </form>
    </div>
  </nav>
</div>
`;
